### PR TITLE
Update `pre-commit` for GitHub Action

### PR DESCRIPTION
**Pull request by michplunkett**
`pre-commit` is now running instead of only `ruff`.

### CODE REVIEW

File: .github/workflows/lint-test.yml [hunk 1]
```diff
@@ -42,12 +42,19 @@ jobs:
...
       run: poetry run pytest
 
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - uses: pre-commit/action@v3.0.0
```
Comment (michplunkett) [inline .github/workflows/lint-test.yml:55]: This makes sure you are running pre-commit in whatever form it's configured to.

**Comment by abejburton** [comment]
I went through and edited all files here so that they pass pre-commit linting and formatting. 

@andrewjtdunn @jchristensonCHI There are functions that are marked as too complex that I added the # noqa: C901 marker to (so that they would pass the linters) that could be good to consider updating if there is time for a refactor.
`civiclens/collect/access_api_data.py:100:5: C901 `pull_reg_gov_data` is too complex (17 > 10)
civiclens/regulations/views.py:23:5: C901 `search_results` is too complex (12 > 10)`

@andrewjtdunn @GregoryHo88 I made some changes to avoid line length issues that broke some test assert statements. Will you look into the failing tests on this branch to see if it would be easy to change them? Ideally we wouldn't have to ignore line length so the linter passes, but we can do that if adjusting the tests is complicated.

After this is merged to main, anyone that pulls these updates should be able to have their linting pass, and pre-commit hooks will be run automatically on pushes and PRs as well as when a commit is made (if you've installed precommit). This will avoid any future formatting issues blocking PRs especially as we ramp up the pace of coding here at the end.

**Comment by jchristensonCHI** [comment]
@abejburton  Just a fyi, the function that the linter is flagging in views.py is the one I specifically showed to James on Friday that he approved and said looked good (the minor changes he recommended of adding a decorator and possibly changing the order of an if/else statement would not fix this)

**Comment by abejburton** [comment]
@jchristensonCHI great, we'll leave that function with # noqa then

**Review by abejburton — APPROVED** [review]

**Pull request merged**